Implement onAdViewCallback

ashbressler opened the pull request:
Please take a look and let me know if this looks good. I did some tests and it's working ok for me. However I have not updated any documentation or examples, which I will do if you think this looks good and is something you would like implemented.

ko2ic commented on lib/banner.dart:
```diff
@@ -138,8 +149,9 @@ class _DFPBannerViewController {
+  final void Function(DFPBannerViewController controller) onAdViewCreated;
 
-  _DFPBannerViewController({
+  DFPBannerViewController({
```
I want the constructor to be private because of not to let they make an instance by mistake.

```
DFPBannerViewController._internal
```

ashbressler replied:
Done

ko2ic commented on lib/banner.dart:
```diff
@@ -55,7 +57,13 @@ class DFPBanner extends StatelessWidget {
-        onPlatformCompleted: (_DFPBannerViewController controller) => controller.load(),
+        onAdViewCreated: onAdViewCreated,
+        onPlatformCompleted: (DFPBannerViewController controller) {
+          controller.load();
```
I want to distinguish the role of the method.

Here we call the _init() method

```
controller._init();
```


At the time of reload call the ```reload()``` method.

Here is a new signature:

```
  Future<void> _init() async {
    _channel.setMethodCallHandler((call) {
      switch (call.method) {
        case "onAdLoaded":
          onAdLoaded();
          break;
        case "onAdFailedToLoad":
          var map = call.arguments.cast<String, int>();
          onAdFailedToLoad(map['errorCode']);
          break;
        case "onAdOpened":
          onAdOpened();
          break;
        case "onAdClosed":
          onAdClosed();
          break;
        case "onAdLeftApplication":
          onAdLeftApplication();
          break;
      }
    });
    return _load();
  }

  Future<void> reload() async {
    return _load();
  }

  Future<void> _load() {
    return _channel.invokeMethod('load', {
      "isDevelop": this.isDevelop,
      "testDevices": this.testDevices?.values,
      "adUnitId": this.adUnitId,
      "adSizes": [this.adSize.toString()],
      "widths": [this.adSize.width],
      "heights": [this.adSize.height],
    });
  }
```

ashbressler replied:
Done

ashbressler commented:
Ok, just pushed some good changes.

I updated the BannerView as per your request and added the documentation to the README.

I added a very detailed sample app, I hope it's not too much, but I found it very useful for myself so I hope you like it.

I figured out why events listeners were not being fired on iOS, it seems like some of the callback methods have changed and needed to be passed 'GAD' objects instead of 'DFP' objects. Take a look at the changes to BannerView.swift and InterstitialAd.swift. I figured it out by looking at RewardedAd.swift and seeing what was different.

ko2ic commented:
I'm sorry for the late review because I forgot my mac on vacation.  I haven't confirmed this yet, but I think it would be nice to read the source.  It will be confirmed by next Tuesday.

Thank you very much for fixing the bug in the listener.

ashbressler commented:
No problem, I'm using my local fork so I've been fine with my development.

Thanks for looking this over.
